dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedFileReader.java

```diff
@@ -59,31 +57,21 @@ public class PagedFileReader extends BlockReader implements PositionReader {
    *
    * @param conf
    * @param cacheManager
-   * @param ufsClient
+   * @param ufs
    * @param fileId
    * @param ufsPath
    * @param fileSize
    * @param startPosition
    * @return a new {@link PagedFileReader}
    */
   public static PagedFileReader create(AlluxioConfiguration conf, CacheManager cacheManager,
-                                       UfsManager.UfsClient ufsClient, String fileId,
+                                       UnderFileSystem ufs, String fileId,
                                        String ufsPath, long fileSize, long startPosition) {
     FileId fileIdField = FileId.of(fileId);
-    CloseableResource<UnderFileSystem> ufs = ufsClient.acquireUfsResource();
-    try {
-      return new PagedFileReader(ufs, LocalCachePositionReader.create(cacheManager,
-          new CloseableSupplier<>(() -> ufs.get().openPositionRead(ufsPath, fileSize)),
-          fileIdField, fileSize, conf.getBytes(PropertyKey.WORKER_PAGE_STORE_PAGE_SIZE),
-          CacheContext.defaults()), fileSize, startPosition);
-    } catch (Throwable t) {
-      try {
-        ufs.close();
-      } catch (Throwable t1) {
-        t.addSuppressed(t1);
-      }
-      throw t;
-    }
```

Comment:
i was wondering what's these multiple try catch wants to gurad against in the first place?

Reply:
Every UnderFileSystem inherits from `BaseUnderFileSystem` which can throw in its `close` method. https://github.com/Alluxio/alluxio/blob/eee6255ecff96d1dc787906340ecb3f1215726ce/dora/core/common/src/main/java/alluxio/underfs/BaseUnderFileSystem.java#L87-L93